### PR TITLE
VPN-2452 - Subscription Expiring Message

### PR DESCRIPTION
**Pull request by jps-moz**
## Description

-Implemented a message addon to notify users when their subscription is expiring.
-They also get an OS notification (until background notifications for mobile is implemented, this will only be for Desktop).
-The ‘Subscription expiring’ message doesn’t show anymore once the subscription end date > 7 days again.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-2542

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

**Review by flodolo — APPROVED** [review]

File: addons/message_subscription_expiring/isExpiring.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(function(api, condition) {
+  //Show message only if within 1 week of expiring.
+  let weekBeforeExpireSecs = (api.subscriptionData.expiresOn / 1000) - 7*24*60*60;
```
Comment (bakulf) [inline addons/message_subscription_expiring/isExpiring.js:3]: technically `expiresOn` can be 0. So, `weekBeforeExpireSecs` will be -604800, which is less than `Date.now()`.

**Review by bakulf — APPROVED** [review]

**Review by bakulf — CHANGES_REQUESTED** [review]
Add a functional test for this addon to simulate its visibility. See the addon folder in tests/tunctional

File: addons/message_subscription_expiring/isExpiring.js (hunk 1 above)
Comment (jps-moz) [reply addons/message_subscription_expiring/isExpiring.js:3]: The other side of that && expression should handle that case :+1:

**Comment by jps-moz** [comment]
> Add a functional test for this addon to simulate its visibility. See the addon folder in tests/tunctional

After exploring this folder and the surrounding test folders, I was unable to find a clear way to trigger the new message directly (for example, by editing the expiresOn property). It seems like these tests are for confirming new functionality or types of addons altogether, which this doesn't fit IMO. I would essentially be duplicating the code, but not testing the one seen by the user. Is there something I am misunderstanding here?

**Pull request closed without merging**